java/system-test/system-test-drivers/src/test/java/sleeper/systemtest/drivers/statestore/StateStoreCommitterLogEntriesTest.java

```diff
@@ -45,11 +47,42 @@ void shouldCountNoCommits() {
         StateStoreCommitterLogEntries logs = new StateStoreCommitterLogEntries(List.of());
 
         // When / Then
-        assertThat(logs.getNumCommitsByTableId())
+        assertThat(logs.countNumCommitsByTableId(Set.of("test-table")))
                 .isEqualTo(Map.of());
     }
 
+    @Test
+    void shouldIgnoreCommitForOtherTable() {
+        // Given
+        StateStoreCommitterLogEntries logs = new StateStoreCommitterLogEntries(List.of(
+                commitToTable("table-1"), commitToTable("table-1"),
+                commitToTable("table-2")));
+
+        // When / Then
+        assertThat(logs.countNumCommitsByTableId(Set.of("table-1")))
+                .isEqualTo(Map.of("table-1", 2));
+    }
+
+    @Test
+    void shouldComputeOverallCommitsPerSecond() {
+        // Given
+        StateStoreCommitterLogEntries logs = new StateStoreCommitterLogEntries(List.of(
+                runStartedAtTime(Instant.parse("2024-08-15T16:17:00Z")),
+                commitToTableAtTime("test-table", Instant.parse("2024-08-15T16:17:01Z"))));
+
+        assertThat(logs.computeOverallCommitsPerSecondByTableId(Set.of("test-table")))
+                .isEqualTo(Map.of("test-table", 1.0));
+    }
+
     StateStoreCommitSummary commitToTable(String tableId) {
         return new StateStoreCommitSummary("test-stream", tableId, "test-commit", Instant.now());
     }
+
```

Comment:
Do we want test that gets close to query limits ?
Downside is generating that much junk data for a single test.

Reply:
For the behaviour of this class, I think 2 or 3 log entries is functionally equivalent to 10,000? I think adding more data would just make the test harder to read?

The code under test isn't coupled to any limiting of the query, so I don't think we'd test that here anyway? Paging behaviour is added in the stacked PR.